### PR TITLE
docs: dfns new signer

### PR DESCRIPTION
**Pull request by avarobinson**
# Pull Request Checklist

- [x] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [x] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [x] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

### Detailed summary
- Added a new page for "Dfns" in the site navigation
- Removed specific wallet signers from the "Signers" section in the introduction page and replaced them with a link to the "Choosing a Signer" page
- Updated the "Supported Signers" page to remove specific wallet signers and added a link to the "Choosing a Signer" page
- Added a new file for the Dfns integration guide
- Added code snippets for initializing and using the Dfns signer in the Dfns integration guide
- Updated the "Why Account Kit" page to remove specific wallet signers and added a link to the "Choosing a Signer" page
- Updated the code example in the "Use it with Light Account" section in the Dfns integration guide to use the `createDfnsSigner` function
- Added new imports and code for the Dfns signer in the `snippets/dfns.ts` file

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: site/smart-accounts/signers/guides/dfns.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    - name: description
+      content: Guide for using Dfns as a signer
+  - - meta
+    - property: og:description
+      content: Guide for using Dfns as a signer
+---
+
+# Dfns Integration Guide
+
+[Dfns](https://www.dfns.co) is an MPC/TSS Wallet-as-a-Service API/SDK provider. We have optimized the balance of security and UX by deploying key shares into a decentralized network on the backend while enabling wallet access via biometric open standards on the frontend like Webauthn. [Ping us](https://www.dfns.co/learn-more) to set up a sandbox environment to get started.
```
Comment (avasisht23) [inline site/smart-accounts/signers/guides/dfns.md:17]:
```suggestion
[Dfns](https://www.dfns.co) is an MPC/TSS Wallet-as-a-Service API/SDK provider. Dfns aims to optimize the balance of security and UX by deploying key shares into a decentralized network on the backend while enabling wallet access via biometric open standards on the frontend like Webauthn. Reach out [here](https://www.dfns.co/learn-more) to set up a sandbox environment to get started.
```

File: site/smart-accounts/signers/guides/dfns.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
+---
+outline: deep
+head:
+  - - meta
+    - property: og:title
+      content: Dfns
+  - - meta
+    - name: description
+      content: Guide for using Dfns as a signer
```
Comment (avasisht23) [inline site/smart-accounts/signers/guides/dfns.md:9]:
```suggestion
      content: Guide for using Dfns as a Signer
```

File: site/smart-accounts/signers/guides/dfns.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  - - meta
+    - property: og:description
+      content: Guide for using Dfns as a signer
```
Comment (avasisht23) [inline site/smart-accounts/signers/guides/dfns.md:12]:
```suggestion
      content: Guide for using Dfns as a Signer
```

File: site/smart-accounts/signers/guides/dfns.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+---
+
+# Dfns Integration Guide
+
+[Dfns](https://www.dfns.co) is an MPC/TSS Wallet-as-a-Service API/SDK provider. We have optimized the balance of security and UX by deploying key shares into a decentralized network on the backend while enabling wallet access via biometric open standards on the frontend like Webauthn. [Ping us](https://www.dfns.co/learn-more) to set up a sandbox environment to get started.
+
+Dfns seamlessly integrates with Account Abstraction by signing User Operations. We've created a full example of a gasless transaction via a paymaster [in our SDK](https://github.com/dfns/dfns-sdk-ts/tree/m/examples/viem/alchemy-aa-gasless) adapted from Alchemy's [sponsor gas example](https://accountkit.alchemy.com/guides/sponsoring-gas.html). Clone the repo to get started and see the comments in the [Readme](https://github.com/dfns/dfns-sdk-ts/tree/m/examples/viem/alchemy-aa-gasless/README.md).
```
Comment (avasisht23) [inline site/smart-accounts/signers/guides/dfns.md:19]:
```suggestion
Dfns seamlessly integrates with Account Abstraction by signing `UserOperation`s. We've created a full example of a gas-less transaction via a paymaster [in our SDK](https://github.com/dfns/dfns-sdk-ts/tree/m/examples/viem/alchemy-aa-gasless), adapted from Alchemy's [gas sponsorship example](https://accountkit.alchemy.com/guides/sponsoring-gas.html). Clone the repo to get started and see the comments in the [Readme](https://github.com/dfns/dfns-sdk-ts/tree/m/examples/viem/alchemy-aa-gasless/README.md).
```

File: site/smart-accounts/signers/guides/dfns.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+```bash [yarn]
+yarn add @dfns/lib-viem @dfns/sdk @dfns/sdk-keysigner
+```
+
+:::
+
+### Create a Dfns SmartAccountSigner
+
+Setup the Dfns Web3 Provider and wrap it in an AlchemyProvider. See the Dfns example [Readme](https://github.com/dfnsext/typescript-sdk/blob/m/examples/viem/alchemy-aa-gasless/README.md) for detail on populating the environment variables.
```
Comment (avasisht23) [inline site/smart-accounts/signers/guides/dfns.md:37]:
```suggestion
Setup the Dfns Web3 Provider and wrap it in an AlchemyProvider. See the Dfns example [Readme](https://github.com/dfnsext/typescript-sdk/blob/m/examples/viem/alchemy-aa-gasless/README.md) for details on populating the environment variables.
```

File: site/smart-accounts/signers/guides/dfns.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+  });
+
+  return DfnsWallet.init({
+    walletId,
+    dfnsClient,
+    maxRetries: 10,
+  });
+};
+
+const alchemyProvider = async (): Promise<AlchemyProvider> => {
```
Comment (avasisht23) [inline site/smart-accounts/signers/guides/dfns.md:77]:
```suggestion
const createAlchemyProvider = async (): Promise<AlchemyProvider> => {
```

File: site/smart-accounts/signers/guides/dfns.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+      })
+  );
+};
+```
+
+### Execute a Gasless Transaction
+
+```ts
+const main = async () => {
+  const provider = await alchemyProvider();
```
Comment (avasisht23) [inline site/smart-accounts/signers/guides/dfns.md:101]:
```suggestion
  const provider = await createAlchemyProvider();
```

File: site/smart-accounts/signers/guides/dfns.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+  const address = await provider.getAddress();
+  console.log(`Smart account address: ${address}`);
+
+  // link the provider with the Gas Manager. This ensures user operations
+  // sent with this provider get sponsorship from the Gas Manager.
+  provider.withAlchemyGasManager({
+    policyId: process.env.ALCHEMY_GAS_POLICY_ID!,
+  });
+
+  const alchemyToken = "0x6F3c1baeF15F2Ac6eD52ef897f60cac0B10d90C3";
```
Comment (avasisht23) [inline site/smart-accounts/signers/guides/dfns.md:111]:
```suggestion
  const nftContractAddress = "0x6F3c1baeF15F2Ac6eD52ef897f60cac0B10d90C3";
```

File: site/smart-accounts/signers/guides/dfns.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+
+  // send a sponsored user operation to mint some tokens
+  const { hash: uoHash } = await provider.sendUserOperation({
+    target: alchemyToken,
```
Comment (avasisht23) [inline site/smart-accounts/signers/guides/dfns.md:115]:
```suggestion
    target: nftContractAddress,
```

File: site/smart-accounts/signers/guides/dfns.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+import { DfnsApiClient } from "@dfns/sdk";
+import { AsymmetricKeySigner } from "@dfns/sdk-keysigner";
+import dotenv from "dotenv";
+import { encodeFunctionData, parseAbi } from "viem";
+import { toAccount } from "viem/accounts";
+import { sepolia } from "viem/chains";
+
+dotenv.config();
+
+const initDfnsWallet = (walletId: string) => {
```
Comment (avasisht23) [inline site/smart-accounts/signers/guides/dfns.md:56]: can you make this a snippet and add it to `site/snippets/dfns.ts`? feel free to look at the other signers to see how they do this.

**Review by avasisht23 — COMMENTED** [review]
nice, just some small changes before a stamp.

can you make sure there's nowhere else they need to add themselves?

for instance, I see signers add themselves to `site/smart-accounts/overview.md` as well.

File: site/snippets/dfns.ts [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+  });
+
+  return DfnsWallet.init({
+    walletId,
+    dfnsClient,
+    maxRetries: 10,
+  });
+};
+
+export const dfnsSigner = async (): Promise<SmartAccountSigner> => {
```
Comment (avasisht23) [inline site/snippets/dfns.ts:37]: nit: function = verb.
```suggestion
export const createDfnsSigner = async (): Promise<SmartAccountSigner> => {
```

File: site/smart-accounts/signers/guides/dfns.md [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+} from "@alchemy/aa-accounts";
+import { encodeFunctionData, parseAbi } from "viem";
+import { sepolia } from "viem/chains";
+import { createDfnsSigner } from "./dfns";
+
+// Remember to replace "ALCHEMY_API_KEY" with your own Alchemy API key, get one here: https://dashboard.alchemy.com/
+const ALCHEMY_API_KEY = null;
+const chain = sepolia;
+
+const createDfnsAlchemyProvider = async (): Promise<AlchemyProvider> => {
```
Comment (avasisht23) [inline site/smart-accounts/signers/guides/dfns.md:61]: un-label the provider to be inline with other examples
```suggestion
const provider = async (): Promise<AlchemyProvider> => {
```

File: site/smart-accounts/signers/guides/dfns.md [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+import {
+  LightSmartContractAccount,
+  getDefaultLightAccountFactoryAddress,
+} from "@alchemy/aa-accounts";
+import { encodeFunctionData, parseAbi } from "viem";
+import { sepolia } from "viem/chains";
+import { createDfnsSigner } from "./dfns";
+
+// Remember to replace "ALCHEMY_API_KEY" with your own Alchemy API key, get one here: https://dashboard.alchemy.com/
+const ALCHEMY_API_KEY = null;
```
Comment (avasisht23) [inline site/smart-accounts/signers/guides/dfns.md:58]:
```suggestion
const ALCHEMY_API_KEY = "API_KEY";
```

**Review by avasisht23 — DISMISSED** [review]
last few nits. 

also, do you want to add them to overview.md or anything else? otherwise, feel free to ship after these last changes.

**Comment by avarobinson** [comment]
> last few nits.
> 
> also, do you want to add them to overview.md or anything else? otherwise, feel free to ship after these last changes.

Don't think signers are added individually anywhere else? Don't see this in overview.md

File: site/smart-accounts/signers/choosing-a-signer.md [hunk 14]
```diff
@@ -48,18 +48,7 @@ The Signer plays a crucial role in your app because it controls the user’s sma
...
-- [web3auth](/smart-accounts/signers/guides/web3auth)
-- [Turnkey](/smart-accounts/signers/guides/turnkey)
-- [Privy](/smart-accounts/signers/guides/privy)
-- [Dynamic](/smart-accounts/signers/guides/dynamic)
-- [Fireblocks](/smart-accounts/signers/guides/fireblocks)
-- [Portal](/smart-accounts/signers/guides/portal)
-- [Capsule](/smart-accounts/signers/guides/capsule)
-- [Lit Protocol](/smart-accounts/signers/guides/lit)
-- [Self-custodial wallets like MetaMask or Ledger](/smart-accounts/signers/guides/eoa)
+Account Kit is compatible with any EIP-1193 provider. Many of the most popular signers can be configured in minutes through our integration signer guides below.
```
Comment (avasisht23) [inline site/smart-accounts/signers/choosing-a-signer.md:51]:
```suggestion
Account Kit is compatible with any EIP-1193 provider. Many of the most popular signers can be configured in minutes through our integration signer guides.
```

File: site/overview/why-account-kit.md [hunk 15]
```diff
@@ -65,9 +65,9 @@ Streamline your sign up flow with simple web2 login options supported by Account
 - Self-custodial wallets like MetaMask or Ledger
 - and more
 
-Account Kit integrates all the leading wallet Signers with bespoke integration guides for [Magic](/smart-accounts/signers/guides/magic), [web3auth](/smart-accounts/signers/guides/web3auth), [Turnkey](/smart-accounts/signers/guides/turnkey), [Privy](/smart-accounts/signers/guides/privy), [Dynamic](/smart-accounts/signers/guides/dynamic), [Fireblocks](/smart-accounts/signers/guides/fireblocks), [Portal](/smart-accounts/signers/guides/portal), [Capsule](/smart-accounts/signers/guides/capsule) and [Lit Protocol](/smart-accounts/signers/guides/lit). Account Kit even supports self-custodial wallets like MetaMask or Ledger. Users can even change their Signer later via Account Kit’s [ownership transfer](/tutorials/transferring-ownership) functionality.
+Account Kit integrates all the leading wallet Signers with bespoke integration guides that you can find [here](/smart-accounts/signers/choosing-a-signer). Account Kit even supports self-custodial wallets like MetaMask or Ledger. Users can even change their Signer later via Account Kit’s [ownership transfer](/tutorials/transferring-ownership) functionality.
```
Comment (avasisht23) [inline site/overview/why-account-kit.md:68]:
```suggestion
Account Kit integrates all the leading wallet Signers with integration guides that you can find [here](/smart-accounts/signers/choosing-a-signer). Account Kit even supports self-custodial wallets like MetaMask or Ledger. Users can even change their Signer later via Account Kit’s [ownership transfer](/tutorials/transferring-ownership) functionality.
```

**Review by moldy530 — APPROVED** [review]

**Pull request merged**